pkg/schema/factory_test.go

```diff
@@ -140,29 +156,57 @@ func TestSchemaCache(t *testing.T) {
 }
 
 func runSchemaTest(t *testing.T, config schemaTestConfig, lookup *mockAccessSetLookup, collection *Collection, testUser user.Info) {
+	resourceID := config.resourceID
+	if resourceID == "" {
+		resourceID = "testCRD"
+	}
+
 	for _, verb := range config.permissionVerbs {
-		lookup.AddAccessForUser(testUser, verb, k8sSchema.GroupResource{Group: testGroup, Resource: "testCRD"}, "*", "*")
+		gr := k8sSchema.GroupResource{
+			Group:    testGroup,
+			Resource: resourceID,
+		}
+		if resourceID == "namespaces" {
+			gr.Group = ""
+		}
+		lookup.AddAccessForUser(testUser, verb, gr, "*", "*")
+	}
+
+	// some of our mock impls panic if the user has literally no access at all.
+	// seed with a harmless one if the test didn't give us any:
+	if len(config.permissionVerbs) == 0 {
+		lookup.AddAccessForUser(testUser, "get", k8sSchema.GroupResource{
+			Group:    testGroup,
+			Resource: "dummy-resource-for-test",
+		}, "*", "*")
+	}
+
+	// pick schema for this test
+	if resourceID == "namespaces" {
+		collection.schemas = map[string]*types.APISchema{"namespaces": makeNamespaceSchema()}
+	} else {
+		collection.schemas = map[string]*types.APISchema{"testCRD": makeSchema("testCRD")}
 	}
 
-	collection.schemas = map[string]*types.APISchema{"testCRD": makeSchema("testCRD")}
 	userSchemas, err := collection.Schemas(testUser)
 	if config.errDesired {
 		assert.Error(t, err, "expected error but none was found")
 	}
 	var testSchema *types.APISchema
 	for schemaName, userSchema := range userSchemas.Schemas {
-		if schemaName == "testCRD" {
+		if schemaName == resourceID || (resourceID == "testCRD" && schemaName == "testCRD") {
 			testSchema = userSchema
```

Comment:
I know it's not part of this PR, but why not break after this line, since it looks like you found the schema you care about. Otherwise it needs a comment why we're continuing to walk the list.